src/components/documentManager.ts

```diff
@@ -0,0 +1,512 @@
+            if (item.BackdropImageTags && item.BackdropImageTags.length) {
+                // get first backdrop of image if applicable
+                src = JellyfinApi.createUrl(
+                    `Items/${item.Id}/Images/Backdrop/0?tag=${item.BackdropImageTags[0]}`
+                );
+            } else if (
+                item.ParentBackdropItemId &&
+                item.ParentBackdropImageTags &&
+                item.ParentBackdropImageTags.length
+            ) {
+                // otherwise get first backdrop from parent
+                src = JellyfinApi.createUrl(
+                    `Items/${item.ParentBackdropItemId}/Images/Backdrop/0?tag=${item.ParentBackdropImageTags[0]}`
+                );
+            }
+        }
+
+        const element: HTMLElement | null = document.querySelector(
+            '#waiting-container-backdrop'
+        );
+
+        if (element === null) {
+            console.error(
+                'documentManager: Cannot find #waiting-container-backdrop'
+            );
+        } else {
+            (<HTMLElement>element).style.backgroundImage =
+                src != null ? 'url(' + src + ')' : '';
+        }
+    }
+
+    // Set a random backdrop, done every backdropPeriodMs.
+    // This function is a tiny bit different from
+    // the old one, it requires that you call getSecurityHeaders yourself.
+    // We could redo this by having an API class that stored
+    // the token and userid along with an ajax proxy to add those headers.
+    private static setRandomUserBackdrop() {
+        const url: string = JellyfinApi.createUserUrl('Items');
+
+        JellyfinApi.authAjax(url, {
+            dataType: 'json',
+            query: {
+                SortBy: 'Random',
+                IncludeItemTypes: 'Movie,Series',
+                ImageTypes: 'Backdrop',
+                Recursive: true,
+                Limit: 1,
+                // Although we're limiting to what the user has access to,
+                // not everyone will want to see adult backdrops rotating on their TV.
+                MaxOfficialRating: 'PG-13'
+            }
+        }).then(function (result) {
+            DocumentManager.setWaitingBackdrop(result.Items[0]);
+        });
+    }
+
+    // Backdrop timer
+    public static clearBackdropInterval() {
+        if (this.backdropTimer !== 0) {
+            clearInterval(this.backdropTimer);
+            this.backdropTimer = 0;
+        }
+    }
+
+    // Start backdrop interval, restart if running, just stop if disabled
+    public static startBackdropInterval() {
+        // avoid running it multiple times
+        this.clearBackdropInterval();
+
+        // skip out if it's disabled
+        if (this.backdropPeriodMs === null) return;
+
+        this.setRandomUserBackdrop();
+
+        this.backdropTimer = <any>setInterval(function () {
+            DocumentManager.setRandomUserBackdrop();
+        }, this.backdropPeriodMs);
+    }
+
+    // null disables backdrops
```

Comment:
TODO here is to clear the backdrop if it gets set to null. Relatively minor issue though.